### ISSUE TITLE
revision stuks in PROGRESS if one of the pipelines fails

### ISSUE THREAD

**Issue by zelima** [datahq/specstore#33]
We have revision stuck in PROGRESS if one of the pipelines fails. 

## Acceptance Criteria

* [x] revision status is `FAILED` if failed

## Tasks 

* [x] failing test
* [x] debug
* [x] fix the bug 

## Analysis 

Problem is that status for one of the pipelines is never updated and always remains pending. For some reasons, PipelineStatusCallback is never called for that one of the pipeline. Eg this is the log from live spec store and PipelineStatusCallback is never called for `derived/zip`:
```
[specstore-1]2018-02-22T13:15:57.313006213Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb: INPROGRESS (errors#=0, stats=None)
[specstore-1]2018-02-22T13:16:01.353616960Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb_csv: INPROGRESS (errors#=0, stats=None)
[specstore-1]2018-02-22T13:16:03.470360039Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/validation_report: INPROGRESS (errors#=0, stats=None)
[specstore-1]2018-02-22T13:16:05.291361552Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb_json: INPROGRESS (errors#=0, stats=None)
[specstore-1]2018-02-22T13:16:07.154292091Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb_csv: FAILED (errors#=12, stats=None)
[specstore-1]2018-02-22T13:16:09.230241456Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb: SUCCESS (errors#=0, stats={'.dpp': {'out-datapackage-url': 'https://s3.amazonaws.com/pkgstore-testing.datahub.io/81429cbbddcfb180f54c142fac32f83b/100kb/100kb/datapackage.json'}, 'bytes': 101247, 'count_of_rows': None, 'dataset_name': '100kb', 'hash': '6dd83e6a5e47b973cb50dc8a4fa710d7'})
[specstore-1]2018-02-22T13:16:11.072389798Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/100kb_json: FAILED (errors#=12, stats=None)
[specstore-1]2018-02-22T13:16:13.155397256Z INFO    :Status ./81429cbbddcfb180f54c142fac32f83b/100kb/1/validation_report: SUCCESS (errors#=0, stats={'.dpp': {'out-datapackage-url': 'https://s3.amazonaws.com/pkgstore-testing.datahub.io/81429cbbddcfb180f54c142fac32f83b/100kb/validation_report/datapackage.json'}, 'bytes': 276897, 'count_of_rows': None, 'dataset_name': '100kb-report', 'hash': '5ab04451bcd7f3da70d604445fd87513'})
```

Debugging further: 

I see there are three of them actually, and all of them are pipelines with dependencies. Seems something wrong in updating dependant pipelines, but can not reproduce locally... 

Hm... seems like I have an older version of planner. Maybe that's a reason why it works locally? 
yes, finally reproduced locally.

Not sure why but if I roll back this two lines, everything seems working fine:
* https://github.com/datahq/planner/commit/5eeb00603d20be054b8262a62e68a7ba20e680d2#diff-2b2385b61acc8c9ecd8b6f72d6b25d77R88
* https://github.com/datahq/planner/commit/5eeb00603d20be054b8262a62e68a7ba20e680d2#diff-2b2385b61acc8c9ecd8b6f72d6b25d77R99
Basically removing `./` from pipeline id

Ok so this is why we can not update pipelines: https://github.com/datahq/specstore/blob/master/flowmanager/controllers.py#L281
So it's comparing 81429cbbddcfb180f54c142fac32f83b/100/8/100kb_json and ./81429cbbddcfb180f54c142fac32f83b/100/8/100kb_json
```
 if dep['pipeline'] == pipeline_id:
     cb(queued_pipeline.pipeline_id, ....)
```
So instead of rolling back the changes in planner, I'm gonna just check `dep['pipeline'].lstrip('./') == pipeline_id`

**Comment by zelima** [comment]
FIXED
